src/main/java/com/bobocode/blyznytsia/bibernate/persister/EntityPersisterImpl.java

```diff
@@ -0,0 +1,162 @@
+package com.bobocode.blyznytsia.bibernate.persister;
+
+import static com.bobocode.blyznytsia.bibernate.util.EntityUtil.getEntityIdValue;
+import static com.bobocode.blyznytsia.bibernate.util.EntityUtil.getEntityNonIdValues;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildDeleteStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildInsertStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildSelectStatement;
+import static com.bobocode.blyznytsia.bibernate.util.SqlUtil.buildUpdateStatement;
+
+import com.bobocode.blyznytsia.bibernate.exception.PersistenceException;
+import com.bobocode.blyznytsia.bibernate.lambda.StatementConsumer;
+import com.bobocode.blyznytsia.bibernate.lambda.StatementFunction;
+import com.bobocode.blyznytsia.bibernate.util.EntityUtil;
+import com.bobocode.blyznytsia.bibernate.util.ResultSetMapper;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+public class EntityPersisterImpl implements EntityPersister {
+
+  Connection connection;
+  ResultSetMapper resultSetMapper;
+
+  public EntityPersisterImpl(Connection connection, ResultSetMapper resultSetMapper) {
+    this.connection = connection;
+    this.resultSetMapper = resultSetMapper;
+  }
+
+  @Override
+  public <T> List<T> findAll(Class<T> entityType, Field key, Object value) {
+    var entities = new ArrayList<T>();
+    var statementText = buildSelectStatement(entityType, key);
+    return performWithinStatement(statementText, false, statement -> {
```

Comment:
[Optional] really don't like this magic true/false as a method params. Is it possible to avoid this ? either by creating another method with nice naming and calling  performWithinStatement with true as generateKeys or just do a local variable.
The same is true about statement lambda expression, could you extract it to a method, so it would be more readable?
but it is up to you

Reply:
* Added `method performWithinStatementWithGeneratedKeys`,  no magic true/false anymore.

* What comes to lamdas - tried to make them as separate functions, but since they consume params from the top-level method, it doesn't improve readability that much, but slightly increases complexity due to 4 extra methods with seemingly similar namings. So I would say it's a necessary evil.

Reply:
Ok, I see. Let's keep it simple then